pkg/policies/policies.go

```diff
@@ -16,61 +16,111 @@
 package policies
 
 import (
-	"bufio"
-	"bytes"
 	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"os"
 	"path/filepath"
 
 	"github.com/bufbuild/protovalidate-go"
+	intoto "github.com/in-toto/attestation/go/v1"
+	"github.com/rs/zerolog"
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
 	"github.com/sigstore/cosign/v2/pkg/blob"
 	"google.golang.org/protobuf/encoding/protojson"
 
 	v1 "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
-	"github.com/chainloop-dev/chainloop/internal/attestation/crafter"
 	v12 "github.com/chainloop-dev/chainloop/internal/attestation/crafter/api/attestation/v1"
-	"github.com/chainloop-dev/chainloop/internal/casclient"
+	"github.com/chainloop-dev/chainloop/internal/attestation/crafter/materials"
 	"github.com/chainloop-dev/chainloop/pkg/policies/engine"
 	"github.com/chainloop-dev/chainloop/pkg/policies/engine/rego"
 )
 
 type PolicyVerifier struct {
-	state *v12.CraftingState
-	cas   casclient.Downloader
+	schema *v1.CraftingSchema
+	logger *zerolog.Logger
 }
 
-func NewPolicyVerifier(state *v12.CraftingState, client casclient.Downloader) *PolicyVerifier {
+func NewPolicyVerifier(schema *v1.CraftingSchema, logger *zerolog.Logger) *PolicyVerifier {
 	// only Rego engine is currently supported
-	return &PolicyVerifier{state: state, cas: client}
+	return &PolicyVerifier{schema: schema, logger: logger}
 }
 
-// Verify verifies that the statement is compliant with the policies present in the schema
-func (pv *PolicyVerifier) Verify(ctx context.Context) ([]*engine.PolicyViolation, error) {
-	violations := make([]*engine.PolicyViolation, 0)
-	policies := pv.state.GetInputSchema().GetPolicies()
-	for _, policyAtt := range policies {
-		if policyAtt.Disabled {
-			// policy is disabled
-			// TODO: WARN.
-			continue
+// VerifyMaterial applies all required policies to a material
+func (pv *PolicyVerifier) VerifyMaterial(ctx context.Context, material *v12.Attestation_Material, artifactPath string) ([]*v12.Policy, error) {
+	result := make([]*v12.Policy, 0)
```

Comment:
I have a problem with the naming here, it seems that the term `Policy` is being overused for both the actual policy and for evaluations of the policy on a material for example.

Wouldn't make sense to change the name of this one to `PolicyEvaluation`?

Reply:
Makes all sense. I'll create a specific struct for this.